storage/storage_reference_test.go

```diff
@@ -0,0 +1,100 @@
+package storage
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStorageReferenceTransport(t *testing.T) {
+	newStore(t)
+	ref, err := Transport.ParseReference("busybox")
+	require.NoError(t, err)
+	transport := ref.Transport()
+	st, ok := transport.(*storageTransport)
+	require.True(t, ok)
+	assert.Equal(t, *(Transport.(*storageTransport)), *st)
+}
+
+func TestStorageReferenceDockerReference(t *testing.T) {
+	ref, err := Transport.ParseReference("busybox")
+	require.NoError(t, err)
+	dr := ref.DockerReference()
+	require.NotNil(t, dr)
+	assert.Equal(t, "busybox:latest", dr.String())
+
+	ref, err = Transport.ParseReference(sha256digestHex)
+	require.Error(t, err)
```

Comment:
`Test…ReferenceDockerReference` is testing `DockerReference`; testing that this value is rejected should happen in `TestTransportParseStoreReference` instead.